### PR TITLE
Update reftests in manifest when reftests become non-reftests

### PR DESCRIPTION
**Pull request by foolip**
In https://github.com/w3c/web-platform-tests/pull/9523 some tests were
converted from reftests to testharness, but the incremental manifest
update mechanism doesn't handle this case. This appears to be the cause
of the mysterious Travis failures on that PR.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/9598)

### CODE REVIEW

**Review by jgraham — APPROVED** [review]

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2018-02-21 10:48:58
Finished: 2018-02-21 11:02:03


<h2>Failing Jobs</h2>
<ul>

<li>tools_unittest in py27</li>

<li>tools_unittest in py36</li>

<li>tools_unittest in pypy</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/24644)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/344257668)

**Review by jgraham — APPROVED** [review]

File: tools/manifest/manifest.py [hunk 1]
```diff
@@ -91,6 +91,8 @@ def update(self, tree):
                     hash_changed = True
                 else:
                     new_type, manifest_items = old_type, self._data[old_type][rel_path]
+                if old_type == "reftest" and new_type != old_type:
```
Comment (gsnedders) [inline tools/manifest/manifest.py]: This might need to include "reftest_node" too?

Comment (foolip) [reply tools/manifest/manifest.py]: I actually don't quite understand what "reftest_node" is. Does it contain all of files involved in a reftest, everything that's not used a ref, or something else? What kind of renaming would one have to do to run into a problem here?

Comment (gsnedders) [reply tools/manifest/manifest.py]: If we had:

a.html: `<link rel=match href=b.html>`
b.html: `<link rel=match href=c.html>`
c.html: `hello world`

Then: 

a.html would be a reftest (it's a test we want to run)
b.html would be a reftest_node (it's in the reftest graph and can itself be run)
c.html would be a support (because we can't tell anything about it statically)

@jgraham I think that's right?

Comment (jgraham) [reply tools/manifest/manifest.py]: Yeah. I guess it's like theoretically possible you have A==B==C and you update B to be a testharness test but still use it as the ref for A, but that seems pretty unlikely, so this case might not get hit in practice, so it's a bit of an edge case correctness issue. File a bug and mark it "good first bug".

Comment (gsnedders) [reply tools/manifest/manifest.py]: I think a less hypothetical case is if b.html changes to `support`? Filing an issue now.

Comment (gsnedders) [reply tools/manifest/manifest.py]: https://github.com/w3c/web-platform-tests/issues/9671
